File: rentacar-frontend/src/services/brandService.ts
import axiosInstance from './axiosInstance';
import { type BrandResponse, type CreateBrandRequest, type UpdateBrandRequest, type PageResponse } from '../models/types';

const BrandService = {
    getAll: async (): Promise<BrandResponse[]> => {
        const response = await axiosInstance.get<PageResponse<BrandResponse>>('/brands');
        return response.data.content;
    },

    getById: async (id: number): Promise<BrandResponse> => {
        const response = await axiosInstance.get<BrandResponse>(`/brands/${id}`);
        return response.data;
    },

    add: async (request: CreateBrandRequest): Promise<void> => {
        await axiosInstance.post('/brands', request);
    },

    update: async (request: UpdateBrandRequest): Promise<void> => {
        await axiosInstance.put('/brands', request);
    },

    delete: async (id: number): Promise<void> => {
        await axiosInstance.delete(`/brands/${id}`);
    }
};

export default BrandService;

File: rentACar/src/main/java/com/sinan/rentACar/business/abstracts/BrandService.java
package com.sinan.rentACar.business.abstracts;


import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.sinan.rentACar.business.requests.CreateBrandRequest;
import com.sinan.rentACar.business.requests.UpdateBrandRequest;
import com.sinan.rentACar.business.responses.GetAllBrandsResponse;
import com.sinan.rentACar.business.responses.GetByIdBrandResponse;

public interface BrandService {
    Page<GetAllBrandsResponse> getAll(Pageable pageable);
    GetByIdBrandResponse getById(int id);
    void add(CreateBrandRequest createBrandRequest);
    void update(UpdateBrandRequest updateBrandRequest);
    void delete(int id);
}
